spec/lib/logstash/java_integration_spec.rb

```diff
@@ -0,0 +1,84 @@
+# encoding: utf-8
+require "spec_helper"
+require "logstash/java_integration"
+
+describe "Java integration" do
+
+  context "type equivalence" do
+
+    it "ArrayList should report to be a Ruby Array" do
+      expect(Java::JavaUtil::ArrayList.new.is_a?(Array)).to be_true
+    end
+
+    it "ArrayList should be class equivalent to Ruby Array" do
+      expect do
+        case Java::JavaUtil::ArrayList.new
+        when Array
+          true
+        else
+          raise
+        end
+      end.not_to raise_error
+    end
+
+    it "LinkedHashMap should report to be a Ruby Hash" do
+      expect(Java::JavaUtil::LinkedHashMap.new.is_a?(Hash)).to be_true
+    end
+
+    it "LinkedHashMap should be class equivalent to Ruby Hash" do
+      expect do
+        case Java::JavaUtil::LinkedHashMap.new
+        when Hash
+          true
+        else
+          raise
+        end
+      end.not_to raise_error
+    end
+  end
```

Comment:
no. these 2 tests using `is_a?` and using `case/when` are exactly & specifically what need to be be supported and tested. Here rspec dsl will make thing **not as clear** because the java integration code has been specifically written to enable both the `is_a?` and `case/when` functionalities on the related Java classes.